code/modules/admin/region_allocator_panel.dm

```diff
@@ -0,0 +1,96 @@
+/// admin panel front end for interacting with the region allocator system
+/datum/region_allocator_panel
+
+/datum/region_allocator_panel/ui_state(mob/user)
+	return tgui_admin_state.can_use_topic(src, user)
+
+/datum/region_allocator_panel/ui_status(mob/user)
+	return tgui_admin_state.can_use_topic(src, user)
+
+/datum/region_allocator_panel/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "RegionAllocatorPanel")
+	ui.open()
+
+/datum/region_allocator_panel/ui_data(mob/user)
+	. = list()
+	for (var/datum/weakref/region_ref as anything in region_allocator.allocated_regions)
+		var/datum/allocated_region/region = region_ref.deref()
+		. += list(list(
+			"ref" = ref(region_ref),
+			"name" = region.name,
+			"x" = region.bottom_left.x,
+			"y" = region.bottom_left.y,
+			"z" = region.bottom_left.z,
+			"width" = region.width,
+			"height" = region.height,
+		))
+	. = list("regions" = .)
```

Comment:
This line's kinda weird. I get that it works, but using `.` as a temporary var as well as a return var in the same proc is probably not good practice.

Reply:
I can just define a temp list and make it a little more readable